Cargo.toml

```diff
@@ -26,6 +27,40 @@ sha1 = { version = "0.10.1", optional = true }
 serde_json = "1.0.61"
 
 [lints.rust]
+rust_2018_idioms = { level = "warn", priority = -1 }
+semicolon_in_expressions_from_macros = "warn"
 unexpected_cfgs = { level = "warn", check-cfg = [
     'cfg(ruma_unstable_exhaustive_types)', # set all types as exhaustive
 ] }
+unreachable_pub = "warn"
+unused_import_braces = "warn"
+unused_qualifications = "warn"
+
+[lints.clippy]
+branches_sharing_code = "warn"
+cloned_instead_of_copied = "warn"
+dbg_macro = "warn"
+disallowed_types = "warn"
+empty_line_after_outer_attr = "warn"
+# exhaustive_enums = "warn"
+# exhaustive_structs = "warn"
```

Comment:
I didn't enable those because they trigger several warnings, and I am not sure if it is wanted here.

Reply:
Probably yes?

Reply:
I added those lints and changed the types then.